src/libstd/thread/mod.rs

```diff
@@ -367,6 +367,26 @@ impl Builder {
 /// want to specify the stack size or the name of the thread, use this API
 /// instead.
 ///
+/// As you can see in the signature of `spawn` there are two constraints on
+/// both the closure given to `spawn` and its return value, let's explain them:
+///
+/// - The `'static` constraint means that the closure and its return value
+///   must have a lifetime of the whole program execution. The reason for this
+///   is that threads can `detach` and outlive the lifetime they have been
+///   created in.  
```

Comment:
Thread detaching is only an explanation for why closure itself is `'static`, and not its return value. The return value has the `'static` bound because it outlives the thread it is returned from; this is true regardless of whether the thread gets detached or not.